src/System.Private.CoreLib/src/System/Array.cs

```diff
@@ -265,8 +271,27 @@ public static void Copy(Array sourceArray, long sourceIndex, Array destinationAr
         // Sets length elements in array to 0 (or null for Object arrays), starting
         // at index.
         //
-        [MethodImplAttribute(MethodImplOptions.InternalCall)]
-        public static extern void Clear(Array array, int index, int length);
+        public static unsafe void Clear(Array array, int index, int length)
+        {
+            if (array == null)
+                ThrowHelper.ThrowArgumentNullException(ExceptionArgument.array);
+
+            ref byte p = ref GetRawArrayGeometry(array, out uint numComponents, out uint elementSize, out int lowerBound, out bool containsGCPointers);
+
+            if (index < lowerBound || index - lowerBound < 0 || length < 0 || (uint)(index - lowerBound + length) > numComponents)
```

Comment:
This is a relatively complex error check and I am having difficulty understanding it. Can you add a comment explaining it. Specifically, `(uint)(index - lowerBound + length) > numComponents`

Reply:
I have added a helper local var to make it more self-explanatory.